rust/worker/src/segment/metadata_segment.rs

```diff
@@ -1178,16 +1182,7 @@ mod test {
                             // Uninitialized segment is fine and means that the record
                             // segment is not yet initialized in storage.
                             RecordSegmentReaderCreationError::UninitializedSegment => None,
-                            RecordSegmentReaderCreationError::BlockfileOpenError(_) => {
-                                panic!("Error creating record segment reader");
-                            }
-                            RecordSegmentReaderCreationError::InvalidNumberOfFiles => {
-                                panic!("Error creating record segment reader");
-                            }
-                            RecordSegmentReaderCreationError::DataRecordNotFound(_) => {
-                                panic!("Error creating record segment reader");
-                            }
-                            RecordSegmentReaderCreationError::UserRecordNotFound(_) => {
+                            _ => {
```

Comment:
I think we should not panic here / can we use this as a moment to clean up

Reply:
this line is inside a test; were you referring to something else?